lib/rucio/core/replica.py

```diff
@@ -2348,31 +2457,64 @@ def update_replicas_states(replicas, nowait=False, *, session: "Session"):
 
         values = {'state': replica['state']}
         if replica['state'] == ReplicaState.BEING_DELETED:
-            query = query.filter_by(lock_cnt=0)
+            stmt = stmt.where(
+                models.RSEFileAssociation.lock_cnt == 0
+            )
             # Exclude replicas use as sources
-            stmt = exists(1).where(and_(models.RSEFileAssociation.scope == models.Source.scope,
-                                        models.RSEFileAssociation.name == models.Source.name,
-                                        models.RSEFileAssociation.rse_id == models.Source.rse_id))
-            query = query.filter(not_(stmt))
+            sub_stmt = select(1).where(and_(models.RSEFileAssociation.scope == models.Source.scope,
```

Comment:
Can we stack these to match the other select statement formats?

Reply:
Changed to:

```
            stmt = stmt.where(
                and_(models.RSEFileAssociation.lock_cnt == 0,
                     not_(exists(select(1)
                                 .where(and_(models.RSEFileAssociation.scope == models.Source.scope,
                                             models.RSEFileAssociation.name == models.Source.name,
                                             models.RSEFileAssociation.rse_id == models.Source.rse_id)))))
            )
```